### ISSUE TITLE
Why does NLLLoss and CrossEntropyLoss require type long?

### ISSUE THREAD

**Issue by fostiropoulos** [pytorch/pytorch#45540]
## 🚀 Feature

Flexible data type of input tensor to NLLLoss and CrossEntropyLoss. As long as it is of type int, the precision is specific to the problem. 

## Motivation

Building a model and defining loss for a multi-class prediction such as for semantic segmentation or reconstruction, if the resolution of the output tensor / image the amount of memory required is absurd. Thus it leads to a significantly smaller batch size. For example, for semantic segmentation of an image 256,256,3 and of 255 classes, the amount of memory required for a batch size larger than 16 and datatype long, doesn't fit in any commercial grade GPU. Moreover for most classification tasks the datatype of long is an overkill. 

## Pitch

NLLLoss(dtype=torch.uint8)
